src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Stream.cs

```diff
@@ -73,8 +82,20 @@ public static Task SerializeAsync(Stream utf8Json, object? value, Type inputType
             return WriteAsyncCore<object>(utf8Json, value!, inputType, options, cancellationToken);
         }
 
-        private static async Task WriteAsyncCore<TValue>(Stream utf8Json, TValue value, Type inputType, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        private static async Task WriteAsyncCore<TValue>(
+            Stream utf8Json,
+            TValue value,
+            Type inputType,
+            JsonSerializerOptions? options,
+            CancellationToken cancellationToken)
         {
+            // We flush the Stream when the buffer is >=90% of capacity.
+            // With the default buffer size of 16K, if a single JSON value is >=1.6K it is possible for
+            // the buffer to max out before it can be flushed, causing the buffer to expand.
+            // So this threshold is a compromise between buffer utilization and minimizing cases where the
+            // buffer is maxed out (and expanded) before it can be flushed.
+            const float FlushThreshold = .9f;
```

Comment:
extreme nit. I only found one other place we don't put a 0 in front of a decimal (`^(?!.*//).* \.\d.*$`)

Reply:
eh, I"ll resolve this one 😄